pkg/client/unversioned/client.go

```diff
@@ -114,6 +115,12 @@ func (c *Client) ComponentStatuses() ComponentStatusInterface {
 	return newComponentStatuses(c)
 }
 
+// WatchOptions is passed as a parameter to per-resource Watch method.
+type WatchOptions struct {
```

Comment:
Technically ListOptions already covers this - we should actually be switching to:

1. Use the list endpoint for watch (remove Prefix("watch") from watchers)
2. Use VersionedParams on request.go with ListOptions which already has resourceVersion
3. Add timeout to ListOptions
4. Take ListOptions on Watch (list options has field and labels selector and resource version already).

Some of those might be more work, but that actually aligns and covers a number of other open issues we have.